src/resource/resource.ts

```diff
@@ -176,6 +180,29 @@ export function isPodOwner(
   return podOwner === webId;
 }
 
+/**
+ *  Given an solidDataset, fetch the urls of all contained resources.
+ *  If the solidDataset given is not a container, or is missing resourceInfo, throw an error.
+ *
+ * @param solidDataset The container from which to fetch all contained resource urls
+ * @returns A list of urls, each of which points to a contained resource of the given solidDataset
+ * @since 1.2.0
+ */
+
+export function getContainedResourceUrlAll(
+  solidDataset: SolidDataset & WithResourceInfo
+): UrlString[] {
+  const contains = "http://www.w3.org/ns/ldp#contains";
+  const container = getThing(solidDataset, getSourceIri(solidDataset));
+  if (container) {
+    return getIriAll(container, contains);
+  } else {
+    // I'm making the assumption here that a dataset is a container if and only if it has a thing at
+    // its root with metadata about the resources contained - this might not be accurate
```

Comment:
It is :) https://www.w3.org/TR/2015/REC-ldp-20150226/#h-ldpc-http_post

> a containment triple MUST be added to the state of the LDPC whose subject is the LDPC URI, whose predicate is ldp:contains and whose object is the URI for the newly created document (LDPR)

```suggestion
```

That said, I would simply return an empty array - the given Resource does not contain other Resources. We already have `isContainer()` to check if it's also a Container (might be useful to mention that in the API docs though).